File: LMS/src/main/java/com/cg/lms/service/AdminService.java
package com.cg.lms.service;

import java.util.List;


import org.springframework.http.ResponseEntity;

import com.cg.lms.entity.User;
import com.cg.lms.model.UserDTO;
public interface AdminService {
	
	public List<UserDTO> getAllUsers(); 
	public List<UserDTO> getAllStudents(String student);
	public List<UserDTO> getAllInstructors(String instructor);
	public List<UserDTO> getAllLibrarians(String librarian);
    public UserDTO getUserById(String userId);   
    public UserDTO getUserByEmail(String email);
    public User createUser(UserDTO userdto);
    public List<User> createMultipleUsers(List<UserDTO> userdto);
    public User updateUserById(String userId, UserDTO userdto);
    public ResponseEntity<String> deleteUserById(String userId);   	   
}

File: LMS/src/main/webapp/lms/src/components/Admin/AdminService.js
import axios from 'axios';
const USER_API_BASE_URL = "http://localhost:8083/user";
class AdminService{
    getAllUsers(){
        return axios.get(USER_API_BASE_URL+"/getAllUsers");
    }

    getAllStudents(){
        return axios.get(USER_API_BASE_URL+"/getAllStudents");
    }
    getAllInstructors(){
        return axios.get(USER_API_BASE_URL+"/getAllInstructors");
    }
    getAllLibrarians(){
        return axios.get(USER_API_BASE_URL+"/getAllLibrarians");
    }

    getUserById(userId){
        return axios.get(USER_API_BASE_URL + "/getUserById/" + userId);
    }
    createUser(user){
        return axios.post(USER_API_BASE_URL+"/createUser/", user);
    }
    updateUser(user, userId){
        return axios.put(USER_API_BASE_URL + "/updateUser/" + userId, user);
    }

    deleteUser(userId){
        return axios.delete(USER_API_BASE_URL + "/deleteUserById/" + userId);
    }
}

export default new AdminService()